src/lib.rs

```diff
@@ -56,6 +56,7 @@
 #![warn(clippy::needless_continue)]
 #![warn(clippy::path_buf_push_overwrite)]
 #![warn(clippy::range_plus_one)]
+#![feature(type_ascription)]
```

Comment:
this is a nightly feature, why do you need it?

Reply:
otherwice `tpl: &(usize, usize)` queue errors.
```type ascription is experimental

note: see issue #23416 <https://github.com/rust-lang/rust/issues/23416> for more information
help: add `#![feature(type_ascription)]` to the crate attributes to enablerustc(E0658)```
Don't know how to get around that

Reply:
you do not need to do that, rust is not swift :)

Reply:
My bad)